src/nvim/api/tabpage.c

```diff
@@ -4,15 +4,51 @@
 #include <stdbool.h>
 #include <stdlib.h>
 
+#include "nvim/api/private/converter.h"
 #include "nvim/api/private/defs.h"
 #include "nvim/api/private/helpers.h"
 #include "nvim/api/tabpage.h"
 #include "nvim/api/vim.h"
 #include "nvim/buffer_defs.h"
+#include "nvim/eval/typval.h"
+#include "nvim/eval/typval_defs.h"
+#include "nvim/eval/typval_encode.h"
+#include "nvim/eval/window.h"
 #include "nvim/globals.h"
 #include "nvim/memory.h"
 #include "nvim/window.h"
 
+/// Returns the layout of a tabpage as a tree-structured Dictionary
+///
+/// @param tabpage Tabpage handle, or 0 for the current tabpage
+/// @param[out] err Error details, if any.
+/// @return Tree of windows and frames in `tabpage`, or nil if the tab is invalid
+Object nvim_tabpage_get_layout(Tabpage tabpage, Error *err)
```

Comment:
```suggestion
Array nvim_tabpage_get_layout(Tabpage tabpage, Error *err)
```

Reply:
What about the nil case? Should I just return an empty array?

Reply:
yes, empty array should be fine (when is that possible ? 😆 )

Reply:
Yes. That's `(Array)ARRAY_DICT_INIT`.

Reply:
Thanks!